ffi/proofs.go

```diff
@@ -221,6 +221,19 @@ func (p *RangeProof) FindNextKey() (*NextKeyRange, error) {
 	return getNextKeyRangeFromNextKeyRangeResult(C.fwd_range_proof_find_next_key(p.handle))
 }
 
+// CodeHashes returns the code hashes contained in all account nodes in the Ethereum trie.
+// This list may contain duplicates and is not guaranteed to be in any particular order.
+// If the proof does not contain any account nodes, an empty list is returned.
+//
+// Note: this method is only relevant for Ethereum tries.
+// This method can only be called after a succesful verification of the range proof,
+// otherwise an error is returned.
+//
+// TODO(#1157): implement this method to extract code hashes from account nodes.
+func (*RangeProof) CodeHashes() ([]Hash, error) {
+	return []Hash{}, nil
+}
```

Comment:
Do we want to return a slice, which requires collecting them all at once, or do we want to return an iterator, which allows lazily collecting them? 

In Go, types that implement [`iter.Seq`](https://pkg.go.dev/iter#Seq) or [`iter.Seq2`](https://pkg.go.dev/iter#Seq2) (for key-value pairs) can be used in range loops making them as ergonomic as rust iterators.